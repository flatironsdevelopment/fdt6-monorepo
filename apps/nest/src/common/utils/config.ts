export const getRedisConfig = (url: string) => {
  const redisUrl = url;
  if (!redisUrl) return {};
  try {
    const { port, password, username, hostname } = new URL(redisUrl);
    return {
      host: hostname,
      port: +port,
      password: password,
      username: username,
    };
  } catch (e) {
    return {};
  }
};
