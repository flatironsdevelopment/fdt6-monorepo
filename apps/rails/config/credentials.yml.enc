EKob807OehkkWhjXcCMdHxGmofKxEIYHVtIvWtghUOdWJ/bhbmJOUA0eJer1o+CTOjVGzAVontoPcyy++tI06aqm58JBu92z5RERkX6Q89bcxY5Zh4cCGILyKDo0aiispE5RMZmWeMcrlVtzZ+QegISoufzHv5enfHstEMQKCP9MUbuchk8J4RCh3TCN0f81xrZKQBgZm9HImt+kb2hHit0EMGx/2fTNhBz5WV1yYrd3g0CMQUdGtEzixMpdTXMz3sD0tFZcTn1fFQid6nq9wu4X2nw8B7tAG4LDaVwUSWga8vSAFJhjwpsDviXfXJzsIqA72FmCY+Fg/+ikivMK+5PlU5yRlupBAo1RbzTkoXnpQHs5mscMOxhk8rsmovV1ITCGECAF2zmQC9p7hxADT8Du6MXJk+pui2Tz--e+43w1shNurGUt7y--ibaNewW21KGXyDl64TuzEw==