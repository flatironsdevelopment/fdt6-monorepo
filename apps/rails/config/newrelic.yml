  common: &default_settings
    license_key: <%= ENV["NEW_RELIC_LICENSE_KEY"] %>
    app_name: <%= ENV["NEW_RELIC_APPLICATION"] %>
    distributed_tracing:
      enabled: true
    agent_enabled: true
    log_level: info

  development:
    <<: *default_settings

  test:
    <<: *default_settings
    monitor_mode: false

  production:
    <<: *default_settings
