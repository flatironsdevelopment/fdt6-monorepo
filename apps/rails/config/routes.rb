# frozen_string_literal: true

require 'sidekiq/web'
Sidekiq::Web.use ActionDispatch::Cookies
Sidekiq::Web.use ActionDispatch::Session::CookieStore, key: '_interslice_session'

Rails.application.routes.draw do # rubocop:todo Metrics/BlockLength
  mount Rswag::Ui::Engine => '/api-docs'
  mount Rswag::Api::Engine => '/api-docs'
  if Rails.env.production?
    Sidekiq::Web.use Rack::Auth::Basic do |username, password|
      ActiveSupport::SecurityUtils.secure_compare(::Digest::SHA256.hexdigest(username),
                                                  ::Digest::SHA256.hexdigest(ENV['SIDEKIQ_USERNAME'])) &
        ActiveSupport::SecurityUtils.secure_compare(::Digest::SHA256.hexdigest(password),
                                                    ::Digest::SHA256.hexdigest(ENV['SIDEKIQ_PASSWORD']))
    end
  end
  mount Sidekiq::Web, at: '/sidekiq'

  get '/health', to: 'health_check#index'

  scope 'api/v1' do
    devise_for :users, controllers: {
      sessions: 'api/v1/users/sessions',
      registrations: 'api/v1/users/registrations',
      confirmations: 'api/v1/users/confirmations',
      passwords: 'api/v1/users/passwords'
    }
  end

  devise_scope :user do
    get 'api/v1/users/_me', to: 'api/v1/users/registrations#show'
    get 'api/v1/users/information', to: 'api/v1/users/informations#show'
    put 'api/v1/users/reset_password', to: 'api/v1/users/reset_passwords#update'
  end

  # Redirect all the non api calls from the frontend
  root to: 'health_check#index'
end
