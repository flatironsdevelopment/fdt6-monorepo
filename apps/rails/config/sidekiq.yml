:verbose: false
:concurrency: 12
:timeout: 60

:queues:
  - default
  - mailers
  - rollbar

production:
  :concurrency: 12
staging:
  :concurrency: 12
development:
  :concurrency: 12
