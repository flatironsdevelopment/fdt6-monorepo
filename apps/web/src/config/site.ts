export type SiteConfig = typeof siteConfig

export const siteConfig = {
  name: 'Flatirons',
  description: "website Description",
  mainNav: [
    {
      title: 'Home',
      href: '/'
    }
  ],
  links: {
  }
}
