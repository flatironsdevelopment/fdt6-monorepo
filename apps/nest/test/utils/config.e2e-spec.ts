import { createConfigMock } from './config';

describe('createConfigMock', () => {
  it('should create a mock configuration object with default values', () => {
    const mockConfig = createConfigMock();

    expect(mockConfig).toHaveProperty('values');
    expect(mockConfig).toHaveProperty('get');
    expect(typeof mockConfig.get).toBe('function');

    expect(mockConfig.values).toMatchSnapshot();

    expect(mockConfig.get('AWS_COGNITO_USER_POOL_ID')).toBe(
      'us-east-2_5DEYLvvFJ',
    );
    expect(mockConfig.get('AWS_REGION')).toBe('us-east-2');
  });

  it('should create a mock configuration object with custom values', () => {
    const customConfigs = {
      AWS_COGNITO_USER_POOL_ID: 'CustomUserPoolID',
      APP_NAME: 'CustomAppName',
    };

    const mockConfig = createConfigMock(customConfigs);

    expect(mockConfig.get('AWS_COGNITO_USER_POOL_ID')).toBe('CustomUserPoolID');
    expect(mockConfig.get('APP_NAME')).toBe('CustomAppName');
  });
});
