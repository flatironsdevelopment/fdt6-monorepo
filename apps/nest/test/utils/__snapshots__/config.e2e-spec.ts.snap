// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createConfigMock should create a mock configuration object with default values 1`] = `
{
  "API_KEY_HEADER": "API_KEY_HEADER",
  "APP_COMPANY": "APP_COMPANY",
  "APP_NAME": "APP_NAME",
  "AWS_COGNITO_CLIENT_ID": "61lhnbrnaj4j0930pevjov22k1",
  "AWS_COGNITO_USER_POOL_ID": "us-east-2_5DEYLvvFJ",
  "AWS_REGION": "us-east-2",
  "HOST_URL": "HOST_URL",
  "PORT": 3000,
  "PREVIEW_EMAILS": "PREVIEW_EMAILS",
  "REDIS_URL": "REDIS_URL",
  "SMTP_FROM": "SMTP_FROM",
  "SMTP_URL": "SMTP_URL",
  "TWILIO_ACCOUNT_SID": "TWILIO_ACCOUNT_SID",
  "TWILIO_AUTH_TOKEN": "TWILIO_AUTH_TOKEN",
  "TWILIO_PHONE_NUMBER": "TWILIO_PHONE_NUMBER",
  "TWILIO_VERIFY_SID": "TWILIO_VERIFY_SID",
}
`;
